### ISSUE TITLE
Problem creating Aurora database in environment level

### ISSUE THREAD

**Issue by jerometremblay** [aws/copilot-cli#5462]
I created a postgresql database using `copilot storage init`. I chose an environment level database.

I have a 'rails' service defined, and it created a `rails/addons/addons.parameters.yml`

```
Parameters:
  ServiceSecurityGroupId: !GetAtt ServiceSecurityGroup.GroupId
```

But when I try to deploy it, i get the following error

```
> copilot deploy
Found only one workload, defaulting to: rails
Environment: development
Checking for all required information. We may ask you some questions.
✘ Proposing infrastructure changes for stack simulations-development-rails
✘ execute deployment 1 of 1 in group 1: deploy service rails to environment development: deploy service: check if changeset is empty: create change set copilot-f12aa4a2-0c54-4629-8671-f5b028627854 for stack simulations-development-rails: ValidationError: Template error: instance of Fn::GetAtt references undefined resource ServiceSecurityGroup
        status code: 400, request id: f87f4514-7d07-49f4-8ca2-463cbae5e2ae: describe change set copilot-f12aa4a2-0b57-4639-8671-f5b028627854 for stack simulations-development-rails: ChangeSetNotFound: ChangeSet [copilot-f12aa4a2-0b57-4639-8671-f5b028627854] does not exist
        status code: 404, request id: a8eb848f-4af4-42d9-983f-9d22e2532e66
```

I tried to inspect the service and the environment generated templates, and I cannot find any `ServiceSecurityGroup` resource.

**Comment by CaptainCarpensir** [comment]
This is failing to deploy because the generated template for the service `rails` is invalid. It looks like somewhere there's a `!GetAtt EnvironmentSecurityGroup` statement in the stack, which would make sense why it's failing because that's an environment stack resource and not a service stack resource. I don't know how this is generated though. Could you post your full environment and service templates here?

**Comment by jerometremblay** [comment]
I pasted the wrong error message at first. I realized it a bit late, and fixed the error message in my original question. 

The real error is 

`Template error: instance of Fn::GetAtt references undefined resource ServiceSecurityGroup`

I had tried to replace it to EnvironmentSecurityGroup manually, I'm not sure I remember why, and copied the wrong output from the console.

The culprit `!GetAtt ServiceSecurityGroup`is in rails/addons/addons.parameters.yml. And this file contains only the two lines I pasted in my initial question.


Attached are the output of
```
copilot svc package --env development --name rails > rails.txt
copilot env package --name development > env.txt
```
[rails.txt](https://github.com/aws/copilot-cli/files/13322837/rails.txt)
[env.txt](https://github.com/aws/copilot-cli/files/13322838/env.txt)

**Comment by jerometremblay** [comment]
Hmm, my rails application is using AppRunner, not a backend service or load-balanced webservice. Maybe that changes something.

**Comment by CaptainCarpensir** [comment]
Ok yes, I think what's happening here is that the RDWS `rails` is not placed inside the VPC, but the Aurora database is created inside the VPC. This means that they can't communicate at all. In order to fix this, you should specify `network.placement: 'private'` in the `rails` manifest.

If this fixes your problem, then we can implement validation to catch this case and throw a more helpful error.

**Comment by jerometremblay** [comment]
> In order to fix this, you should specify `network.placement: 'private'` in the `rails` manifest.

This was indeed the problem, once I uncommented the lines in my manifest to set the vpc, it deployed fine.

One thing that confused me is that usually (often? sometimes?) configuration files like that have the default values in the comment, and uncommenting is equivalent to hard-coding default values. In this case, the values in comments produced a different behavior than the default.

**Comment by Lou1415926** [comment]
@jerometremblay Can I interpret your comment as: you were expecting "placing a Request-Driven Web Service in the VPC" as the default behavior (that you wouldn't need to specify `network.placement: 'private'` to explicitly enable it), and it was surprising to you that the default behavior was to _not_ place it in the VPC 🤔 ?

If what I understand above ⬆️ is correct - I can see why it can be confusing: any of the other architectures (Load-Balanced Web Service, Backend Service, etc.) that Copilot offers by default place the service in the VPC, why is it different for Request-Driven Web Service?

The main reason for this "unique" default is simply that App Runner did not support VPC when it was first launched (also when we first supported it), and also that the App Runner default is to _not_ place it in the VPC, too.

When App Runner started supporting VPC - also when we started supporting placing a RDWS in a VPC - we decided to _not_ make it a default so that we don't break any existing RDWS users. Plus, putting an AppRunner service (thus RDWS) in a VPC can be more costly than the default. We didn't want people to pay the money when they don't need to.

Hope this clarifies things!
